homeworks/week2/hw6.md

```diff
@@ -13,5 +13,17 @@ isValid([3, 5, 8, 13, 22, 35])
 ```
 
 ## 執行流程
-1. ..
-2. ..
\ No newline at end of file
+1. 執行第 12 行 isValid 函式，呼叫 function isValid(arr) 帶入引數 [3, 5, 8, 13, 22, 35] 到函式中判斷
+2. 執行第 3 行，設定變數 i 是 0，檢查 i 是否 <= 陣列的長度，是，繼續執行，開始進入第一圈迴圈
+3. 執行第 4 行，判斷引數的陣列 index[0] 的值是否小於 0 ，否，繼續執行迴圈到 i 的值大於 arr.length的長度，超過長度後執行第四步驟
+4. 執行第 6 行，設定變數 i 是 2，檢查 i 是否 <= 陣列的長度，是，繼續執行，開始進入第一圈迴圈
+5. 執行第 7 行，判斷引數的陣列 index[2] 有沒有不等於 index[1] + index[0] 的值，否
+6. 第一圈迴圈結束，跑回第 6 行，i++，i 變成 3，檢查 i 是否 <= 陣列的長度，是，開始進入第二圈迴圈
+7. 執行第 7 行，判斷引數的陣列 index[3] 有沒有不於 index[2] + index[1] 的值，否
+8. 第二圈迴圈結束，跑回第 6 行，i++，i 變成 4，檢查 i 是否 <= 陣列的長度，是，開始進入第三圈迴圈
+9. 執行第 7 行，判斷引數的陣列 index[4] 有沒有不等於 index[3] + index[2] 的值，是，執行 if 判斷式中的內容
+10. 回傳字串 'invalid' 給 isValid([3, 5, 8, 13, 22, 35])
```

Comment:
因為 10. 這一句話去查了 ECMAScript 2015 對於 `return` 的[描述](https://www.ecma-international.org/ecma-262/6.0/#sec-return-statement) 提到：

A return statement causes a function to cease execution and **return a value to the caller**. If Expression is omitted, the return value is undefined. Otherwise, the return value is the value of Expression.

這樣看來的確是返還給呼叫器，這邊也請 @aszx87410 來印證一下。

Reply:
是，就是這樣，有去查 spec 給推